cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -1001,6 +1001,95 @@ struct Trunc {
   }
 };
 
+// TODO(edponce): Move builders to function state and reset.
+struct Divmod {
+  template <typename T, typename Arg0, typename Arg1>
+  static enable_if_floating_value<T, FixedSizeListScalar> Call(KernelContext* ctx, Arg0 dividend, Arg1 divisor, Status* st) {
+    T quotient = std::floor(dividend / divisor);
+    T remainder = dividend - quotient * divisor;
+
+    NumericBuilder<T> builder;
+    builder.Append(quotient);
+    builder.Append(remainder);
+    std::shared_ptr<Array> array;
+    builder.Finish(&array);
+    FixedSizeListScalar list(array, builder.type())
```

Comment:
Using a FixedSizeListScalar is a really expensive way of doing this.

Reply:
We use a `StructArray` in all such cases, e.g. the Mode kernel.

Reply:
@edponce I don't think it's worth fitting this function into the same framework as the others. Just write it out directly and have it directly build the result StructArray.

Reply:
Well, if you want to figure out the template metaprogramming needed to make a result type of `std::tuple` be written into a `StructArray`… :slightly_smiling_face:

Reply:
Ok, I will revert to a `StructArray`. Thanks!